test/test_profile.py

```diff
@@ -159,3 +163,114 @@ def execute(
             """
         )
         assert result == self.expected_value
+
+
+def test_dump_profile_command(default_params, task_manager_creator):
+    """Test saving the browser profile using a command."""
+    manager_params, browser_params = default_params
+    manager_params.num_browsers = 1
+    manager, _ = task_manager_creator((manager_params, browser_params[:1]))
+    cs = CommandSequence(url=BASE_TEST_URL)
+    cs.get()
+    tar_path = manager_params.data_directory / "profile.tar.gz"
+    cs.dump_profile(tar_path, True)
+    manager.execute_command_sequence(cs)
+    manager.close()
+    assert tar_path.is_file()
+
+
+def test_load_tar_file(tmp_path):
+    """Test that load_profile does not delete or modify the tar file."""
+    tar_path = Path("profile.tar.gz")
+    profile_path = tmp_path / "browser_profile"
+    browser_params = BrowserParamsInternal(browser_id=1)
+    modified_time_before_load = tar_path.stat().st_mtime
+    load_profile(profile_path, browser_params, tar_path)
+    assert modified_time_before_load == tar_path.stat().st_mtime
+
+
+def test_crash_during_init(default_params, task_manager_creator):
+    """Test that no profile is saved when Task Manager initialization crashes."""
+    manager_params, browser_params = default_params
+    manager_params.num_browsers = 1
+    browser_params[0].profile_archive_dir = (
+        manager_params.data_directory / "browser_profile"
+    )
+    # This will cause the browser launch to fail
+    browser_params[0].seed_tar = Path("/tmp/NOTREAL")
+    with pytest.raises(ProfileLoadError):
+        manager, _ = task_manager_creator((manager_params, browser_params[:1]))
+    tar_path = browser_params[0].profile_archive_dir / "profile.tar.gz"
+    assert not tar_path.is_file()
```

Comment:
@vringar, is this what you had in mind in https://github.com/mozilla/OpenWPM/issues/789#issuecomment-817917602 ?

Reply:
Sorry for not getting back to you in time.

I think what I wanted to express there, that if
1. We visit `http://a.com`
2. We crash visiting `b.com`
3. We visit `http://c.com`
all with a stateful crawl on the same browser, then `http://a.com` and `http://c.com` should both be in the `moz_places` DB

Reply:
That sounds like something that's tested by `test_profile_recovery`, no?
`during_init` is set to True [here](https://github.com/mozilla/OpenWPM/blob/b1dbee125a887b23724ec8ab0985cabcdf04b044/openwpm/task_manager.py#L186L188), if the task manager fails to launch the browsers the first time.
So from what I understand, when testing [this condition](https://github.com/mozilla/OpenWPM/blob/b1dbee125a887b23724ec8ab0985cabcdf04b044/openwpm/browser_manager.py#L606) we want to test that because the browsers didn't start successfully and did not execute any commands, no meaningful profile was created and thus no profile archive was saved.

Reply:
You are of course correct!